amaranth/ml/train.py

```diff
@@ -52,24 +57,69 @@ def main():
       low_calorie_threshold=amaranth.LOW_CALORIE_THRESHOLD,
       high_calorie_threshold=amaranth.HIGH_CALORIE_THRESHOLD)
 
+  # Normalize input strings
+  # Step 1: convert strings to lowercase
+  # Step 2: filter out characters present in DISH_NAME_FILTERS
+  # Step 3: re-combine characters back to a normal string
+  calorie_data['description'] = calorie_data['description'].apply(
+      lambda desc: desc.lower(),  # Step 1
+  ).apply(
+      lambda desc: [char for char in desc
+                    if char not in DISH_NAME_FILTERS],  # Step 2
+  ).apply(
+      ''.join,  #Step 3
+  )
+
   # Do some preprocessing and calculations for encoding
-  corpus = calorie_data['description'].str.replace(',', '').str.lower()
+  corpus = calorie_data['description']
   vocab_size = lib.num_unique_words(corpus)
   tokenized_corpus = corpus.map(lambda desc: desc.split(' '))
   max_corpus_length = lib.max_sequence_length(tokenized_corpus)
 
-  # Create TextVectorization layer
-  vectorize_layer = keras.layers.experimental.preprocessing.TextVectorization(
-      max_tokens=vocab_size,
-      output_sequence_length=max_corpus_length,
-  )
-
-  vectorize_layer.adapt(calorie_data['description'].to_numpy())
+  # Encode 'description' into new column 'input'
+  calorie_data['tokenized'] = calorie_data.apply(
+      lambda row: text.one_hot(row['description'], vocab_size), axis=1)
+
+  # Create tokenizer to encode input text
+  tokenizer = dict()  # Dictionary mapping each unique word to a unique int
+  tokenizer_cnt = dict()  # Counts appearances of each word in tokenizer
```

Comment:
Might be worth using defaultdict(int) here. Example: https://docs.python.org/3/library/collections.html#collections.defaultdict

Reply:
Changed in https://github.com/googleinterns/amaranth/pull/35/commits/09c21da0537a006070af43b3301c511759b25d81